server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -139,14 +154,34 @@ def _validate_data_types(self):
             curr_axis = getattr(self.data, str(ax))
             for ann in curr_axis:
                 datatype = curr_axis[ann].dtype
-                downcast_map = {'int64': 'int32',
-                                'uint32': 'int32',
-                                'uint64': 'int32',
-                                'float64': 'float32',
+                downcast_map = {"int64": "int32",
+                                "uint32": "int32",
+                                "uint64": "int32",
+                                "float64": "float32",
                                 }
                 if datatype in downcast_map:
                     warnings.warn(f"Scanpy annotation {ax}:{ann} is in unsupported format: {datatype}. "
                                   f"Data will be downcast to {downcast_map[datatype]}.")
+                if isinstance(datatype, CategoricalDtype):
+                    category_num = len(curr_axis[ann].dtype.categories)
+                    if category_num > 100 and category_num > self.max_category_items:
```

Comment:
I think 100 is too conservative (low), and recommend a higher number.   Perhaps 500.   There isn't much difference on the front-end until we get into the 500-1000 range, so doesn't seem worth the warning at 100.

Reply:
✅